### ISSUE TITLE
Cursor Position Requests (CPR) issues when running IPython on Docker on Mac M1

### ISSUE THREAD

**Issue by davisfung1116** [davisfung1116/Self-Learning-with-the-Book-Python-for-Finance#1]
I am self-learning from the book Python for Finance (2nd Edition) and prepared Dockerfile and install.sh accordingly. Docker Desktop was installed in advance.

The "install.sh" is like: https://github.com/yhilpisch/py4fi2nd/blob/master/code/ch02/Docker/install.sh
(Just minor change on line #20-23: the conda version I used is Miniforge3 which allows Mac M1 running Linux on Docker)

and Dockerfile:
https://github.com/yhilpisch/py4fi2nd/blob/master/code/ch02/Docker/Dockerfile
(Just minor change on line #26 to: ENV PATH /root/miniforge3/bin:$PATH)

I put both files in a folder and execute, in Mac-M1 terminal, 2 commands according to the book.
FIRST:
docker build -t py4fi:basic .
SECOND:
docker run -t py4fi:basic

EVERYTHING WORKED FINE, JUST AT THE END I GET THIS:
Python 3.9.7 | packaged by conda-forge | (default, Oct 10 2021, 15:08:54) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.29.0 -- An enhanced Interactive Python. Type '?' for help.

^[[24;1RWARNING: your terminal doesn't support cursor position requests (CPR).
In [1]: 

CONSEQUENCE - I could type anything, but when press ENTER, it just leads me to another line:
In [1]: print("hello world")
exit()
exit
exit

I try to leave by "exit()", ENTER, "exit", ENTER repeatedly, nothing happen but bringing me to a new line everytime.
I have no choice but press Ctrl+C to leave.

I believe this is related to the warning: "WARNING: our terminal doesn't support cursor position requests (CPR)."

I googled the warning sentence but the answers are something like ... when seeing that sentence, use a different command:
python -i
... but this does not fit OR handling emacs by editing .macs with:
(setq python-shell-interpreter "ipython"
  python-shell-interpreter-args "--simple-prompt")
... this seems related but i am not sure whether it really fits and how to implement.

In short, is there solution which I just need to edit Dockerfile and/or install.sh to let IPython respond properly?

Thanks every respondents in advance, I am a newcomer, excuse me for the bad format.
